MatrixSDK/Background/MXBackgroundStore.swift

```diff
@@ -83,45 +58,177 @@ class MXBackgroundStore: MXMemoryStore {
+    func storeState(forRoom roomId: String, stateEvents: [MXEvent]) {
         //  no-op
     }
     
     //  Fetch real room state
-    override func state(ofRoom roomId: String, success: @escaping ([MXEvent]) -> Void, failure: ((Error) -> Void)? = nil) {
+    func state(ofRoom roomId: String, success: @escaping ([MXEvent]) -> Void, failure: ((Error) -> Void)? = nil) {
         fileStore.state(ofRoom: roomId, success: success, failure: failure)
     }
     
     //  Fetch real soom summary
-    override func summary(ofRoom roomId: String) -> MXRoomSummary? {
+    func summary(ofRoom roomId: String) -> MXRoomSummary? {
         return fileStore.summary(ofRoom: roomId)
     }
     
     //  Fetch real room account data
-    override func accountData(ofRoom roomId: String) -> MXRoomAccountData? {
+    func accountData(ofRoom roomId: String) -> MXRoomAccountData? {
         return fileStore.accountData(ofRoom: roomId)
     }
     
-    //  Override and return a user to be stored on session.myUser
-    override func user(withUserId userId: String) -> MXUser? {
-        if userId == credentials.userId, let myUser = myUser {
-            //  if asking for session user and myUser is set, return that
-            return myUser
-        }
-        return MXUser(userId: userId)
-    }
-    
-    override var syncFilterId: String? {
+    var syncFilterId: String? {
         get {
             return fileStore.syncFilterId
         } set {
             //  no-op
         }
     }
     
+    func event(withEventId eventId: String, inRoom roomId: String) -> MXEvent? {
+        guard let roomStore = roomStore(forRoom: roomId) else {
+            return nil
+        }
+        
+        let event = roomStore.event(withEventId: eventId)
+        
+        NSLog("[MXBackgroundStore] eventWithEventId: \(eventId) \(event == nil ? "not " : "" )found")
+        return event
+    }
+    
+    
+    //  MARK: - Private
+    private func roomStore(forRoom roomId: String) -> MXFileRoomStore? {
+        // Use the cached instance if available
+        if let roomStore = roomsStore[roomId] {
+            return roomStore
+        }
+        
+        guard let roomStore = fileStore.roomStore(forRoom: roomId) else {
+            NSLog("[MXBackgroundStore] roomStore: Unknown room id: \(roomId)")
+            return nil
+        }
+        
+        roomsStore[roomId] = roomStore
+        return roomStore
+    }
+    
+    
+    //  MARK: - Stubs
+    
+    /// Following operations should be not required
+    
+    func open(with credentials: MXCredentials, onComplete: (() -> Void)?, failure: ((Error?) -> Void)? = nil) {
+    }
+    
+    func storeEvent(forRoom roomId: String, event: MXEvent, direction: __MXTimelineDirection) {
+    }
+    
+    func replace(_ event: MXEvent, inRoom roomId: String) {
+    }
+    
+    func eventExists(withEventId eventId: String, inRoom roomId: String) -> Bool {
+        return false
```

Comment:
It'd be perfect to implement this in an efficient way in order to avoid reading whole events for only one. Not mandatory for now but it can be a good improvement point in future.

Reply:
I do not think we use this method in the bg service. If I am wrong, we need to implement it.

Reply:
No, you right, as we're not using it right now. Mine was just a wish.